homework/src/index.js

```diff
@@ -30,17 +34,104 @@
     return elem;
   }
 
-  function main(url) {
-    fetchJSON(url, (err, data) => {
+  function addRow(label, value, tbody) {
+    const tr = createAndAppend('tr', tbody);
+    createAndAppend('td', tr, { class: 'label', text: label });
+    createAndAppend('td', tr, { text: value });
+  }
+
+
+  function renderRepository(info, container) {
+    container.innerHTML = '';
+    const table = createAndAppend('table', container);
+    const tbody = createAndAppend('tbody', table);
+
+    const tr1 = createAndAppend('tr', tbody);
+    createAndAppend('td', tr1, {
+      text: 'Repository: ',
+    });
+    const repository = createAndAppend('td', tr1);
+    createAndAppend('a', repository, {
+      href: info.html_url,
+      target: '_blank',
+      text: info.name
+    });
+
+    const descriptions = `${info.description} `;
+    addRow('Description: ', descriptions, tbody);
+    const forks = `${info.forks} `;
+    addRow('Forks: ', forks, tbody);
+    const updated = `${info.updated_at} `;
+    addRow('Updated: ', updated, tbody);
+
+  }
+
+  function renderContributors(info, container) {
+    container.innerHTML = '';
+    createAndAppend('h3', container, { text: 'Contributors' });
+    const ul = createAndAppend('ul', container);
+
+    info.forEach(contributor => {
+      const li = createAndAppend('li', ul);
+      createAndAppend('img', li, { src: contributor.avatar_url });
+      const a = createAndAppend('a', li, { href: contributor.html_url, target: '_blank' });
+      a.innerText = contributor.login;
+      createAndAppend('p', li, { text: contributor.contributions });
+    });
+  }
+
+  async function main(url) {
+    try {
+      const repositories = await fetchJSON(url);
+
       const root = document.getElementById('root');
-      if (err) {
+      const header = createAndAppend('header', root, { text: 'HYF Repositories' });
+      const select = createAndAppend('select', header);
+      const container = createAndAppend('div', root, { id: 'container' });
+      const leftDiv = createAndAppend('div', container, { id: 'left_div' });
+      const rightDiv = createAndAppend('div', container, { id: 'right_div' });
+
+      repositories.sort(function (a, b) {
+        return a.name.localeCompare(b.name);
+      });
+
+      repositories.forEach((repository, index) => {
+        createAndAppend('option', select, {
+          value: index, text: repository.name
+        });
+      });
+
+      select.addEventListener('change', async () => {
+        const repository = repositories[select.value];
+
+        try {
+          renderRepository(repository, leftDiv);
+          const contributors = await fetchJSON(repository.contributors_url);
+          renderContributors(contributors, rightDiv);
+        }
+        catch (err) {
+          const root = document.getElementById('root');
+          createAndAppend('div', root, { text: err.message, class: 'alert-error' });
+        }
+
+      });
+
+      try {
```

Comment:
This inner `try` is unnecessary (and therefore unwanted) as the outer `try` block that starts in line 86 already deals with catching errors. If you remove lines 121 and 125-129 then any error that could occur in line 123 (for instance, a rejected promise) would be could by the `catch` block of lines 132-135.